### DISCUSSION TITLE
Collapsible threads

### DISCUSSION THREAD

**Discussion by evan2645** [community/community#2901] [Discussions]
The discussions feature is great for organizing and streamlining responses and related conversation, however to me it feels only slightly better than an issue because you still have to scroll past all the conversation(s), and when there are a lot it feels like information overload.

It would be awesome if each individual response/thread was collapsible, perhaps even by default... and at that point, maybe it also makes sense for the creator of a response/thread to be able to give it a short title. I imagine then that the UI could show each response/thread in a discussion in a way similar to how the Gmail inbox renders emails/threads, making it easy to home in on the parts of the discussion that you care about.

**Comment by becca** [comment]
> maybe it also makes sense for the creator of a response/thread to be able to give it a short title

That sounds like a lot of work! What about having the threads NOT being collapsed is creating a lot of pain for you? Do you think it will be painful for you to have to manually unfurl each thread to read through them?

**Reply by evan2645** [reply]
> That sounds like a lot of work!

Heh :) that may be the case, but nothing is free!

> What about having the threads NOT being collapsed is creating a lot of pain for you?

I was hoping to use the discussions feature to drive conversations around large feature proposals. In such proposals, conversation generally occurs around certain aspects or portions of the original issue. Each of those conversations can individually get quite long, and there could be a dozen or more of those conversations occurring on a single proposal.

Most people will be interested in engaging with only one or two of those dozen plus conversations. When the threads are not collapsible, identifying the individual conversations that are important to you is difficult. You need to scroll past many pages of responses in order to find the conversation you're looking for. This also makes it difficult to discover conversations that may be relevant to you.

> Do you think it will be painful for you to have to manually unfurl each thread to read through them?

I'm sure there are all sorts of use cases for the discussions feature, and I am focused on only one... so my opinion is unlikely to represent the full set.

For my use case, engagement with the discussions feature rarely (perhaps never?) involves reading through everything in one pass. So I don't envision having to sit there and unfurl everything - rather, I unfurl the ones I want, the ones I'm looking for, when I'm interested in them. To me, this is exactly the way that I engage with my Gmail inbox. Users starting conversations on the proposal start one per topic, rather than one per user with all the feedback lumped into one.

One option could be that the furled/unfurled default be a user preference, similar to the way that unified/split diff is a user preference.

**Reply by becca** [reply]
How do you know which threads to unfurl? Also, do you have any examples of these conversations right now (are they happening in Issues?) that I could use as a reference?

**Reply by czyzby** [reply]
I think the top-level replies should be visible by default, but their replies could potentially be collapsed and loaded on demand - especially in case of longer threads. This is similar to how Reddit works for example, where most of the replies are hidden by default if there is a lot of comments.

**Reply by evan2645** [reply]
> How do you know which threads to unfurl?

The same way I know which emails to click on in my inbox (and also the reason I suggested the possibility of a title). Under Gmail, the subject is displayed along with a small preview of the beginning of the text. I realize that there would be no search feature in the GitHub case, but still the Gmail inbox interface seems to have done this well.

> Also, do you have any examples of these conversations right now (are they happening in Issues?) that I could use as a reference?

In my case, these conversations are currently happening in issues. They get messy pretty fast. Here is one example: https://github.com/spiffe/spire/issues/1945

Organizing the conversation using topic-based threads in the GitHub discussions feature would make it a lot easier to understand what's going on there, and to have a productive conversation(s) that produces results. Of course, there are other ways that we can solve this problem using non-GH tooling. I'm in no way insinuating that we _must_ solve it using Discussions :). Assuming this is a use case you'd like to target with Discussions, then IMO something like I'm describing is going to be necessary.

> I think the top-level replies should be visible by default, but their replies could potentially be collapsed and loaded on demand - especially in case of longer threads.

This might also work

**Comment by wojpawlik** [comment]
I don't think extra title functionality is necessary. Collapsing could simply show only the beginning of the comment, which might happen to include level 1 or 2 markdown heading.
